android-components/components/feature/fxsuggest/src/main/java/mozilla/components/feature/fxsuggest/facts/FxSuggestFactsMiddleware.kt

```diff
@@ -40,12 +42,14 @@ class FxSuggestFactsMiddleware : Middleware<BrowserState, BrowserAction> {
         context: MiddlewareContext<BrowserState, BrowserAction>,
         action: BrowserAction,
     ) = when (action) {
-        is AwesomeBarAction.EngagementFinished ->
-            if (action.abandoned) Unit else emitSuggestionFacts(context.state.awesomeBarState)
+        is AwesomeBarAction.EngagementFinished -> emitSuggestionFacts(
+            awesomeBarState = context.state.awesomeBarState,
+            engagementAbandoned = action.abandoned,
+        )
```

Comment:
I think [we do call `next(action)`](https://github.com/mozilla-mobile/firefox-android/blob/0cdadb1a02477a1849099377ce85c451037913e7/android-components/components/feature/fxsuggest/src/main/java/mozilla/components/feature/fxsuggest/facts/FxSuggestFactsMiddleware.kt#L35-L36), but in `invoke`, one method up 😊

I borrowed the `handleAction` pattern [from `MetricsMiddleware`](https://github.com/mozilla-mobile/firefox-android/blob/0fd50e31e9665c213d4128703c0382466cb014fa/fenix/app/src/main/java/org/mozilla/fenix/components/metrics/MetricsMiddleware.kt#L23-L24), but `FxSuggestFactsMiddleware` looks a little different because its `handleAction` also takes the context—I wonder if it looks too similar to `invoke` at first glance?